### DISCUSSION TITLE
Deprecated signature for useWindowScroll

### DISCUSSION THREAD

**Discussion by lucpotage** [vueuse/vueuse#2726] [Q&A] [answered]
Hi, I wanted to use `useWindowScroll` composable and got the following error:

```js
const useWindowScroll: ({ window }?: ConfigurableWindow | undefined) => {
    x: Ref<number>;
    y: Ref<number>;
}
@deprecated — Please use usescroll instead.

The signature '({ window }?: ConfigurableWindow | undefined): { x: Ref<number>; y: Ref<number>; }' of 'useWindowScroll' is deprecated.ts(6387)
index.d.ts(4486, 4): The declaration was marked as deprecated here.
```

The [docs](https://vueuse.org/core/usewindowscroll/#usewindowscroll) don’t seem to have deprecated this composable. I don’t know what is wrong.

I also tried useScroll but the values are not reactive.

**Comment by ojvribeiro** [comment]
Yeah, `useScroll` just doesn't work like `useWindowScroll` and the docs doesn't give a hint of how we're supposed to use it.

**Reply by ouuan** [reply]
`useScroll(window)` instead of `useWindowScroll()`?

**Reply by ojvribeiro** [reply]
Makes sense 😅

**Comment by ouuan** [comment]
It seems that the implementation of `useScroll` is much heavier than `useWindowScroll`, so using `useScroll` may lead to a slightly larger bundle size and learning cost. I'd prefer not to deprecate it. Maybe add a note in the documentation of `useWindowScroll` to redirect users to `useScroll` if they want the features provided by `useScroll`.

[the remark this answers is not in the document]
**Reply by okxiaoliang4** [reply]
I think we should keep `useWindowScroll` it's more focused on one thing simple and useful.
